modules/admin-ui/src/main/webapp/scripts/shared/services/wizards/new-series/metadata.js

```diff
@@ -92,7 +92,16 @@ angular.module('adminNg.services')
 
           params.presentableValue = presentableValue;
         } else {
-          params.presentableValue = params.collection[value];
+          if (params.collection.hasOwnProperty(value)) {
+            params.presentableValue = params.collection[value];
+          } else {
+            // this should work in older browsers, albeit looking clumsy
+            var matchingKey = Object.keys(params.collection)
+              .filter(function(key) {return params.collection[key] === value;})[0];
+            params.presentableValue = params.type === 'ordered_text'
+              ? JSON.parse(matchingKey)['label']
+              : matchingKey;
+          }
```

Comment:
Which older browsers is this supposed to support? (As far as I know, the Admin UI doesn't work for Internet Explorer anyway...) Would there be a more elegant solution otherwise? Also if I understand this correctly you assume that the entries of an ordered_text metadata field will always be in JSON, and I'm not sure that has to be that way...

Reply:
I was not aware that the Admin UI does not support Internet Explorer. I sometimes forget that this project has been in existence for a long time, not realising that old looking code probably **is** old.

This would be more elegant (and more performant, for that matter): 
```Object.keys(params.collection).find(key => params.collection[key] === value);```

As to the possibility of the *ordered_text* type not being formatted as a JSON string literal, I think this will (in our case) never happen because for an element to be ordered, there needs to be information about a positional index, as well as a value, which almost requires a JS object. Or am I missing something?